src/lib/pgp-key.c

```diff
@@ -457,7 +457,7 @@ pgp_seckey_t *
 pgp_decrypt_seckey(const pgp_key_t *key, FILE *passfp)
 {
     if (key->key.seckey.decrypt_cb == NULL) {
-        return NULL;
+        return (pgp_seckey_t *) &key->key.seckey;
```

Comment:
Because we properly free the decrypted secret keys now (#350), this will result in a crash.